src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java

```diff
@@ -88,6 +103,81 @@ public TarantoolTemplate(
         );
     }
 
+    private MessagePackMapper getDefaultComplexTypesMapper() {
+        // blocked because it's not deepcopy
+        // MessagePackMapper defaultConverterWithTupleConverter = tarantoolClient.getConfig().getMessagePackMapper().copy();
+        return DefaultMessagePackMapperFactory
+                .getInstance()
+                .defaultComplexTypesMapper();
+    }
+
+    private void registerTupleResultMapper(MessagePackMapper customMapper,
+                                           Optional<TarantoolSpaceMetadata> spaceMetadata) {
+        customMapper.registerValueConverter(
+                ArrayValue.class,
+                TarantoolTuple.class,
+                new TarantoolAutoResultConverter<>(
+                        new DefaultTarantoolTupleValueConverter(customMapper,
+                                spaceMetadata.orElse(null)),
+                        spaceMetadata.orElse(null)));
+    }
+
+    private <T> CallResultMapper<T, SingleValueCallResult<T>>
+    withDefaultSingleValueMapper(MessagePackMapper customMapper,
+                                 Class<T> entityClass) {
+        return mapperFactoryFactory
+                .getDefaultSingleValueMapper(customMapper, entityClass);
+    }
+
+    private <T> CallResultMapper<T, SingleValueCallResult<T>>
+    getAutoResultMapper(Class<T> entityClass, Optional<TarantoolSpaceMetadata> spaceMetadata) {
+        MessagePackMapper customMapper = getDefaultComplexTypesMapper();
+        registerTupleResultMapper(customMapper, spaceMetadata);
```

Comment:
Registering a new mapper will not affect the source mapper since the containers are copied. Don't worry about the nested mappers, they will not be affected by that.
Do you have any tests showing a problem with using the coping ctor here?

Reply:
This is the problem, they are copied superficially, there was a problem that `TarantoolAutoResultConverter `, after adding, appeared in all requests. I'll try to add a test - reproducer

Reply:
```diff
diff --git a/src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java b/src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java
index 8357163..b5c66be 100644
--- a/src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java
+++ b/src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java
@@ -105,10 +105,10 @@ public class TarantoolTemplate extends BaseTarantoolTemplate {

     private MessagePackMapper getDefaultComplexTypesMapper() {
         // blocked because it's not deepcopy
-        // MessagePackMapper defaultConverterWithTupleConverter = tarantoolClient.getConfig().getMessagePackMapper().copy();
-        return DefaultMessagePackMapperFactory
-                .getInstance()
-                .defaultComplexTypesMapper();
+        return tarantoolClient.getConfig().getMessagePackMapper().copy();
+//        return DefaultMessagePackMapperFactory
+//                .getInstance()
+//                .defaultComplexTypesMapper();
     }

     private void registerTupleResultMapper(MessagePackMapper customMapper,
```
If we change to copy, then after a few tests we will see that copying does not occur and `tarantoolClient.getConfig().getMessagePackMapper()`

<img width="598" alt="image" src="https://user-images.githubusercontent.com/22258443/153198594-6223b066-e584-42be-8b25-13acb6f27778.png">

Reply:
I think that creating a new mapper and registering a new converter each time callForSingleResult is called is a bad idea. Looks like useless work. Why is it impossible to create the necessary mappers for all methods in interfaces once and use the ready ones?

Reply:
This is a good point, but it's a connectors problem in general. In particular, a new mapper is now being created for each request in the master branch. There is a ticket for this issue about caching mappers https://github.com/tarantool/cartridge-java/issues/117 . Your idea is interesting, it will be necessary to think whether it is realistic to implement and whether there will be problems. But I propose to try realize it in another PR (we can leave a ticket on whether it is realistic to implement such a task)

Reply:
Discussed with @vrogach2020 that we should leave it as it is, not reuse the custom one until we fix this problem in [cartridge-java#166](https://github.com/tarantool/cartridge-java/issues/166).